.github/workflows/main.yaml

```
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
```

IMO Any branch should run the tests, not just master.

```suggestion
  push:
  pull_request:
```